docs/GettingStarted.md

```diff
@@ -269,4 +276,81 @@ Patches are submitted to LLVM/MLIR via GitHub pull-requests, the basic flow is a
 6) Publish the branch on your fork of the repository and create a GitHub pull-request.
 
 When your review converges and your patch is approved, it can be merged directly on GitHub.
-If you have commit access, you can do this yourself, otherwise a reviewer can do it for you.
+If you have commit access, you can do this yourself, otherwise a reviewer can do it for you.  
+  
+## Writing a basic CIRCT Pass  
+Passes can be added at several levels in CIRCT. Here we illustrate this with a simple pass, added 
+to the `firrtl` dialect, that replaces all of the wire names with `foo`: 
+1) Update your dialect’s Passes.td file to define what your pass will do in a high-level, well documented way, 
+e.g. in `include/circt/Dialect/FIRRTL/Passes.td`:  
+```
+def FooWires : Pass<"firrtl-foo-wires", "firrtl::CircuitOp"> {
```

Comment:
My bad, I think I originally wrote this for hw and then switched to firrtl for some reason. I'll clean everything up in a external file that actually runs to avoid brain farts like this lol